src/transport/tls.transport.ts

```diff
@@ -1,18 +1,34 @@
 import { Transport } from './transport';
 import { TLSSocket, connect } from 'tls';
 import * as URL from 'url';
+import { SocksClient } from 'socks';
 
 export class TlsTransport extends Transport<{ url: string; enableTrace?: boolean }> {
```

Comment:
The `proxyOptions` should be here.

Reply:
Ok! Didn't see that. And where do you think I should put the TransportProxyOptions type? Should I rename it also?

Reply:
> And where do you think I should put the TransportProxyOptions type? Should I rename it also?

These should be in the `tsl.transport.ts` file. You may rename it to `TlsTransportProxyOptions`.